packages/flutter/test/material/text_field_test.dart

```diff
@@ -711,6 +711,43 @@ void main() {
     expect(controller.selection.extentOffset, eIndex);
   });
 
+  testWidgets('Read only text field', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController(text: 'readonly');
+
+    await tester.pumpWidget(
+        overlay(
+          child: TextField(
+            controller: controller,
+            readOnly: true,
+          ),
+        )
+    );
+    // Read only text field cannot open keyboard.
+    await tester.showKeyboard(find.byType(TextField));
+    expect(tester.testTextInput.hasActiveClient, false);
+    await skipPastScrollingAnimation(tester);
+
+    expect(controller.selection.isCollapsed, true);
+
+    // Press the 'd' should not open keyboard.
```

Comment:
nit: maybe specify that you mean the 'd' in readonly. I was confused by this comment... :)